cmake/BuildNghttp2.cmake

```diff
@@ -0,0 +1,73 @@
+#
+# A macro to build the bundled nghttp2 library.
+macro(nghttp2_build)
+    set(NGHTTP2_SOURCE_DIR ${PROJECT_SOURCE_DIR}/third_party/nghttp2)
+    set(NGHTTP2_BINARY_DIR ${PROJECT_BINARY_DIR}/build/nghttp2/work)
+    set(NGHTTP2_INSTALL_DIR ${PROJECT_BINARY_DIR}/build/nghttp2/dest)
+
+    # See BuildLibCURL.cmake for details.
+    set(NGHTTP2_CFLAGS "")
+    if (TARGET_OS_DARWIN AND NOT "${CMAKE_OSX_SYSROOT}" STREQUAL "")
+        set(NGHTTP2_CFLAGS "${NGHTTP2_CFLAGS} ${CMAKE_C_SYSROOT_FLAG} ${CMAKE_OSX_SYSROOT}")
+    endif()
+
+    set(NGHTTP2_CMAKE_FLAGS "-DENABLE_WERROR:BOOL=ON")
```

Comment:
Shouldn't we set the same `CMAKE_BUILD_TYPE` as in tarantool (but don't touch `ENABLE_DEBUG`, it said that it degrades performance significantly)?

See also https://github.com/tarantool/tarantool/issues/4575.

Reply:
Updated:

```diff
    list(APPEND NGHTTP2_CMAKE_FLAGS "-DCMAKE_INSTALL_LIBDIR=lib")
    list(APPEND NGHTTP2_CMAKE_FLAGS "-DCMAKE_INSTALL_INCLUDEDIR=include")
    list(APPEND NGHTTP2_CMAKE_FLAGS "-DCMAKE_INSTALL_BINDIR=bin")

+  list(APPEND NGHTTP2_CMAKE_FLAGS "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
    list(APPEND NGHTTP2_CMAKE_FLAGS "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
```

Reply:
Nice!